packages/next/src/utils.ts

```diff
@@ -1832,7 +1832,7 @@ export const onPrerenderRoute =
     if (nonDynamicSsg || isFallback || isOmitted) {
       outputPathData = outputPathData.replace(
         new RegExp(`${escapeStringRegexp(origRouteFileNoExt)}.json$`),
-        `${routeFileNoExt}.json`
+        `${routeFileNoExt.replace(/\$/g, '$$$$')}.json`
```

Comment:
Maybe good to add a comment explaining:
- what we're doing (the `$$$$` can be confusing 😄)
- why we're doing it (i'm actually not sure about this, why do we need that line here?)